docs/design/validation.md

```diff
@@ -0,0 +1,1989 @@
+
+#### New Handling
+
+This framework can implement separate types and logic for validation. This aligns well with other design decisions in the framework and will enable it to provide targeted solutions that can capture context and functionality appropriately for various validation scenarios, which may not be appropriate if bundled with other functionality such as plan modifications.
+
+### Typed Parameters Versus Request and Response Types
+
+#### Typed Parameters
+
+The framework could implement bespoke error, path, and value types in function parameters and returns. For example:
+
+```go
+func(context.Context, path *tftypes.AttributePath, value attr.Value) error
+```
+
+While very explictly defining function signatures specific to the validation concepts as they are understood today (such as validation currently only being against configuration) to potentially make testing and implementation details easier, this presents future compatibility concerns. Any changes or additions would require breaking changes. Any semantic differences about the context of a call when it reaches the function cannot be captured. The rest of the framework has opted for a request and response type pattern to handle these concerns, where choosing typed parameters here does not seem to provide much benefit to have a separate implementation.
+
+#### Request and Response Types
+
+The framework could implement the request and response pattern for validation, typed to each RPC. For example:
+
+```go
+func(context.Context, ValidateRequest, *ValidateResponse)
+```
+
+Ease of implementation and testing is slightly reduced because of the wrapper types, however compatibility is more guaranteed. The framework can signal smaller deprecations and implement underlying migrations if necessary. Each request and response can be tailed to the exact context and functionality available at the time.
+
+Examples in the rest of the proposals will prefer this style where appropriate.
+
+### Validation Function Types Versus Interfaces
+
+#### Validation Function Type
+
+New Go type(s) could be created that define the signature of a validation function, similar to the previous framework `SchemaValidateFunc`. For example:
+
+```go
+type ValidationFunc func(context.Context, ValidationRequest, *ValidationResponse)
+```
+
+To support passing through the provider instance to the function, the parameters would also need to include a `tfsdk.Provider` interface type:
```

Comment:
Technically it could be included in the request info, I think?

Reply:
I believe it can! I will add that as a possibility too.